src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,41 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+
+/**
+ * Represents a deadline task.
+ */
+public class Deadline extends Task {
+    private LocalDateTime dateTime;
+
+    /**
+     * Creates an instances of Deadline.
+     *
+     * @param description description of task
+     * @param dateTime date and time of task
+     */
+    public Deadline(String description, LocalDateTime dateTime) {
+        super(description);
+        super.taskType = TaskType.D;
+        this.dateTime = dateTime;
+    }
+
+    /**
+     * Get the date and time in the format which match in data file.
+     *
+     * @return string represent the date and time
+     */
+    public String getDateTime() {
+        return dateTime.format(IN_FORMATTER);
+    }
+
+    @Override
+    public String toString() {
+        return "[" + super.getTaskType() + "]" + super.toString() + " (by: " + dateTime.format(OUT_FORMATTER) + ")";
```

Comment:
consider using a String.format so that it is easier to read what the output string is.

Reply:
Thanks, I will change it to a string to increase the readability.